### PR TITLE
Add snapshot release workflow triggered by /snapshot-release

### PR DESCRIPTION
**Pull request by sam-b-rose**
Trigger a [snapshot release](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md) using a slash command in your feature PR (`/snapshot-release`)

### CODE REVIEW

File: .github/workflows/slash-command.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    name: Dispatch
+    runs-on: ubuntu-latest
+    steps:
+      uses: peter-evans/slash-command-dispatch@v3
+      with:
+        token: ${{ secrets.PAT }}
+        permission: admin
+        issue-type: pull-request
+        commands: |
+          snapshot
```
Comment (alex-page) [inline .github/workflows/slash-command.yml]:
```suggestion
        commands: snapshot
```

File: .github/workflows/slash-command.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  slashCommand:
+    name: Dispatch
+    runs-on: ubuntu-latest
+    steps:
+      uses: peter-evans/slash-command-dispatch@v3
+      with:
+        token: ${{ secrets.PAT }}
+        permission: admin
```
Comment (alex-page) [inline .github/workflows/slash-command.yml:16]:
```suggestion
        permission: admin
```

Are we only allowing `admin` to do snapshot releases? I think any Shopifolk should be able to use it.

File: .github/workflows/snapshot.yml [hunk 3]
```diff
@@ -23,21 +23,22 @@ jobs:
...
+          git checkout -b snapshot
+          git reset --hard ${{ github.ref }}
 
       - name: Create release Pull Request or publish to NPM
         id: changesets
         uses: changesets/action@v1
         with:
-          publish: yarn release
+          version: yarn version-packages --snapshot snapshot
+          publish: yarn release --no-git-tags --snapshot
```
Comment (aaronccasanova) [inline .github/workflows/snapshot.yml]:
```suggestion
          version: yarn version-packages -- --snapshot snapshot
          publish: yarn release -- --no-git-tags --snapshot
```

Do we need to add option terminators to ensure the additional flags are passed to the `changeset` command?

File: .github/workflows/snapshot.yml [hunk 4]
```diff
@@ -23,21 +23,22 @@ jobs:
...
         env:
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Remove snapshot branch
+        run: |
+          git checkout ${{ github.ref }}
+          git branch -D snapshot
```
Comment (aaronccasanova) [inline .github/workflows/snapshot.yml]: I know I originally proposed this step, but am now wondering if it's needed. A fresh runner is used each time this workflow is triggered and we never push the temp `snapshot-release` branch. Should we delete?

Comment (alex-page) [reply .github/workflows/snapshot.yml]: 🚮

**Review by alex-page — COMMENTED** [review]
- [ ] Update release documentation
- [ ] Only allow shopifolk to run `/snapshot-release`

File: .github/workflows/snapshot.yml (hunk 4 above)
Comment (sam-b-rose) [reply .github/workflows/snapshot.yml]: Great point! Nice catch

File: .github/workflows/snapshot.yml (hunk 3 above)
Comment (sam-b-rose) [reply .github/workflows/snapshot.yml]: I did test and no extra `--` needed here 👍

**Pull request merged**